### PR TITLE
partitionccl: parallelize slow tests

### PR DESCRIPTION
**Pull request by RaduBerinde**
This patch improves `TestInitialPartitioning` and `TestRepartitioning`
to run tests on 8 parallel servers instead of one, bringing down the
running time for each one to 15-20s instead of 100+s.

Release note: None

Related to #37028.

### CODE REVIEW

**Review by RaduBerinde — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/37773)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @danhhz and @jordanlewis)

___
*[pkg/ccl/partitionccl/partition_test.go, line 1296 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/37773#-Lf_DC36EfqJudk84WgI:-Lf_DC36EfqJudk84WgJ:b-21icvp) ([raw file](https://github.com/cockroachdb/cockroach/blob/2f3e60bfd4564de71b2c8459ed8807e6d3af689d/pkg/ccl/partitionccl/partition_test.go#L1296)):*
> ```Go
> 	const parallelism = 8
> 	var splitTestCases [parallelism][]repartitioningTest
> 	for i := range splitTestCases {
> ```

Before anyone asks why I didn't extract this into a function, it's because the types are different `[]partitioningTest` vs `[]repartitioningTest`.

**Review by jordanlewis — COMMENTED** [review]
:lgtm: thanks! 

But it looks like one of them failed in the race tests? I'm not sure but worth investigation.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/37773#-:-Lf_agdX83jHBhDUmxZj:bsg91tp)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @danhhz and @jordanlewis)

**Comment by RaduBerinde** [comment]
Fantastic, `sqlDB.Exec(t, `CREATE DATABASE data`)` stuck waiting for a reply for 19 minutes. Probably tickled an existing bug..

**Comment by RaduBerinde** [comment]
I reduced the parallelism in race mode, but I'm not super sure of this change anymore. A simpler improvement would be to add a `t.Parallel()` to these big tests so they at least run in parallel with each other. I'll leave it up to @danhhz.

**Review by danhhz — COMMENTED** [review]
Sorry this dropped off my radar for a bit. The complexity is unfortunate and I'd really love to find some time to speed up the tests themselves (even if I have to add some new testing knobs, feels like there should be something we can do there), but I think the incremental win in reduced CI time is worth it. Thanks, Radu!

:lgtm:

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/37773#-:-Lg3TlCm-eeIpK51byyG:ba6q6xt)* status: :shipit: complete! 1 of 0 LGTMs obtained (and 1 stale) (waiting on @danhhz and @RaduBerinde)

___
*[pkg/ccl/partitionccl/partition_test.go, line 1324 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/37773#-Lg3TROA4UMSA5EiGJ5N:-Lg3TROB7cbWSLgQzOCd:b-av159h) ([raw file](https://github.com/cockroachdb/cockroach/blob/07d8e3947d05843efb3a1f4ccf36c6114600f419/pkg/ccl/partitionccl/partition_test.go#L1324)):*
> ```Go
> 
> 				for _, test := range testCases {
> 					t.Run(fmt.Sprintf("%s/%s", test.old.name, test.new.name), func(t *testing.T) {
> ```

If you're feeling ambitious, I'd find this easier to read about if we pulled this whole function into one thing that took a `testing.T` and a `repartitioningTest`. That way the parallel subtest stuff is separate from the test itself

**Review by RaduBerinde — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/37773)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 2 stale)

___
*[pkg/ccl/partitionccl/partition_test.go, line 1324 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/37773#-Lg3TROA4UMSA5EiGJ5N:-LhS7VvqDXZWt54KAlzB:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/07d8e3947d05843efb3a1f4ccf36c6114600f419/pkg/ccl/partitionccl/partition_test.go#L1324)):*
<details><summary><i>Previously, danhhz (Daniel Harrison) wrote…</i></summary><blockquote>

If you're feeling ambitious, I'd find this easier to read about if we pulled this whole function into one thing that took a `testing.T` and a `repartitioningTest`. That way the parallel subtest stuff is separate from the test itself
</blockquote></details>

Done.

**Review by danhhz — APPROVED** [review]
Thanks again!

**Pull request merged**
